management/src/test/java/org/ehcache/management/cluster/ClusteringManagementServiceTest.java

```diff
@@ -234,15 +235,22 @@ private void remotelyUpdateCollectedStatistics() throws Exception {
       assertThat(client, is(notNullValue()));
       final ClientIdentifier ehcacheClientIdentifier = client;
 
+      CountDownLatch callCompleted = new CountDownLatch(1);
       AtomicReference<String> managementCallId = new AtomicReference<>();
       BlockingQueue<ContextualReturn<?>> returns = new LinkedBlockingQueue<>();
 
       agent.setContextualReturnListener(new ContextualReturnListener() {
         @Override
         public void onContextualReturn(ClientIdentifier from, String id, ContextualReturn<?> aReturn) {
-          assertEquals(ehcacheClientIdentifier, from);
-          assertEquals(managementCallId.get(), id);
-          returns.offer(aReturn);
+          try {
+            assertEquals(ehcacheClientIdentifier, from);
+            // make sure the call completed
+            callCompleted.await();
```

Comment:
Please do not have unlimited awaits - instead have a timeout just in case something really bad happens.